prov/verbs/src/verbs_ep.c

```diff
@@ -395,8 +395,8 @@ static int fi_ibv_process_xrc_preposted(struct fi_ibv_srq_ep *srq_ep)
 	struct slist_entry *entry;
 	int ret;
 
-	while (!slist_empty(&srq_ep->prepost_list)) {
-		entry = slist_remove_head(&srq_ep->prepost_list);
+	while (!slist_empty(&srq_ep->xrc.prepost_list)) {
```

Comment:
The locking is off here.  We don't acquire the prepost lock until after we've walked this list and removed items.

Reply:
The lock is obtained in fi_ibv_ep_enable_xrc(), the SRQ is created, the SRQ MSG ops are swapped out so they no longer add to the prepost list, and fi_ibv_process_xrc_preposted() is called to post the actual receives, and the lock is freed.

Reply:
If you agree, I will add a comment that the srq_ep::prepost_lock must be held when invoking fi_ibv_process_xrc_preposted.

Reply:
thanks